include/common/core/memory.hpp

```diff
@@ -233,78 +230,132 @@ constexpr __ESIMD_NS::atomic_op get_atomic_op(gpu::xetla::atomic_op ao) {
-/// vector_size per SIMD channel).
-/// @tparam DS is the data size.
-/// @tparam L1H is L1 cache hint.
-/// @tparam L2H is L2 cache hint.
-/// @tparam N is the number of SIMD channels (platform dependent).
-/// @param p       [in] is the base pointer.
-/// @param offsets [in] is the zero-based offsets in bytes.
-/// @param pred    [in] is predicates.
+/// The next functions perform transposed 1-channel prefetch.
+/// template <typename T, int VS = 1, typename OffsetT,
+///           typename PropertyListT = empty_properties_t>
+/// void prefetch(const T *p, OffsetT byte_offset, simd_mask<1> mask,
+///                   PropertyListT props = {});                   // (usm-pf-7)
+/// void prefetch(const T *p, OffsetT byte_offset,
+///                   PropertyListT props = {});                   // (usm-pf-8)
+/// template <typename T, int VS = 1,
+///           typename PropertyListT = empty_properties_t>
+/// void prefetch(const T *p, simd_mask<1> mask,
+///                   PropertyListT props = {});                   // (usm-pf-9)
+/// void prefetch(const T *p, PropertyListT props = {});           //(usm-pf-10)
 ///
+
+/// template <typename T, int N, int VS, typename OffsetT,
+///           typename PropertyListT = empty_properties_t>
+/// void prefetch(const T *p, simd<OffsetT, N / VS> byte_offsets,
+///                   simd_mask<N / VS> mask,
+///                   PropertyListT props = {});                   // (usm-pf-1)
+/// Supported platforms: DG2, PVC only.
+/// Prefetches elements of the type 'T' from memory locations addressed
+/// by the base pointer \p p and byte offsets \p byte_offsets, to the cache.
+/// Access to any element's memory location can be disabled via the input vector
+/// of predicates \p mask. If mask[i] is unset, then the prefetch from
+/// (p + byte_offsets[i]) is skipped.
+/// @tparam T Element type.
+/// @tparam N Number of elements to read.
+/// @tparam VS Vector size. It can also be read as the number of reads per each
+/// address. The parameter 'N' must be divisible by 'VS'.
+/// @param p The base address.
+/// @param byte_offsets the vector of 32-bit or 64-bit offsets in bytes.
+/// For each i, ((byte*)p + byte_offsets[i]) must be element size aligned.
+/// @param mask The access mask.
+/// @param props The optional compile-time properties. Only cache hint
+/// properties are used.
 template <
-    typename Ty,
-    int NElts = 1,
-    data_size DS = data_size::default_size,
+    typename T,
+    int N,
+    int VS,
     cache_hint L1H = cache_hint::cached,
-    cache_hint L2H = cache_hint::cached,
-    int N>
+    cache_hint L2H = cache_hint::cached>
 __XETLA_API void xetla_prefetch_global(
-    Ty* p,
-    xetla_vector<uint32_t, N> offsets,
-    xetla_mask<N> pred = 1) {
-  using T = native_type_t<Ty>;
+    T* p,
+    xetla_vector<uint32_t, N / VS> byte_offsets,
+    xetla_mask<N / VS> mask = 1) {
+#if __INTEL_LLVM_COMPILER >= 20240200
```

Comment:
any performance tests for the xetla instruction level modification ?

Reply:
For 1x4096x4096 and 32x4096x4096 1204x4096x4096 UT performance is the same.

Reply:
The performance of a single instruction is not easy to test